### PR TITLE
Temp u4 8829 dashboards from remote feed

### PR DESCRIPTION
**Pull request by perploug**
Adds a dashboard to the content section which loads content from dashboard.umbraco.org
Caches content and css in runtime cache.

### CODE REVIEW

File: src/Umbraco.Web/Editors/DashboardController.cs [hunk 1]
```diff
@@ -6,14 +6,114 @@
...
+
+                content = new JObject();
+                ApplicationContext.ApplicationCache.RuntimeCache.InsertCacheItem<JObject>(key, () => content, timeout: new TimeSpan(0, 5, 0));
+            }
+
+            return content;
+        }
+
+        [WebApi.UmbracoAuthorize]
+        public async Task<HttpResponseMessage> GetRemoteDashboardCss(string section, string baseUrl = "http://dashboard.umbraco.org/")
```
Comment (nul800sebastiaan) [inline src/Umbraco.Web/Editors/DashboardController.cs]: This should be `https`, no?

File: src/Umbraco.Web/Editors/DashboardController.cs [hunk 2]
```diff
@@ -6,14 +6,114 @@
...
-    public class DashboardController : UmbracoAuthorizedJsonController
+    [ValidationFilter]
+    [AngularJsonOnlyConfiguration]
+    [IsBackOffice]
+    public class DashboardController : UmbracoApiController
     {
+        //we have baseurl as a param to make previewing easier, so we can test with a dev domain from client side
+        [WebApi.UmbracoAuthorize]
+        [ValidateAngularAntiForgeryToken]
+        public async Task<JObject> GetRemoteDashboardContent(string section, string baseUrl = "http://dashboard.umbraco.org/")
```
Comment (nul800sebastiaan) [inline src/Umbraco.Web/Editors/DashboardController.cs]: Was expecting this to be `https` as well. :)

File: src/Umbraco.Web/Editors/DashboardController.cs (hunk 1 above)
Comment (perploug) [inline src/Umbraco.Web/Editors/DashboardController.cs]: Ah yes, at the time of the PR there was not a SSL cert in place, but there is now, so everything can be https

File: src/Umbraco.Web/Editors/DashboardController.cs (hunk 2 above)
Comment (perploug) [inline src/Umbraco.Web/Editors/DashboardController.cs]: same as above, can be https now

**Comment by nul800sebastiaan** [comment]
Some code review feedback (I've made all the changes already):
- Use descriptive variable names `dashboardTab` instead of `dashTab`, `context`, instead of `ctx` - makes it easier for everyone to read the code
- Never catch `Exception`, always catch a sepecific exception, in this case `HttpRequestException`
- Always log exceptions when you catch them, there's nothing worse than something silently failing and not being able to find why
- `RuntimeCache.GetCacheItem` and `RuntimeCache.InsertCacheItem` can be combined into `RuntimeCache.GetCacheItem` which either gets the item from the cache or puts an item in the cache if it doesn't yet exist
- `HttpClient` needs to be wrapped in a `using` block so it gets cleaned up when done
- Avoid deep nested if statements, instead of `if(something) { if(somethingElse) { // } }` use `if(something == false) { continue; }` and then on the next line you can start a new if without all the indentation.
- `Trim()` removes all spaces, readlines and newlines already, no need to specify them specifically
- If you're doing `ToLower()` always use `ToLowerInvariant()` instead
- There were quite a few Resharper hints that weren't followed, which are some of the notes of this review above

**Pull request merged**